packages/webdriverio/src/commands/element/scrollIntoView.ts

```diff
@@ -2,9 +2,127 @@ import logger from '@wdio/logger'
 import { ELEMENT_KEY } from 'webdriver'
 
 import { getBrowserObject } from '@wdio/utils'
+import type { CustomScrollIntoViewOptions, MobileScrollIntoViewOptions } from 'src/types.js'
 
 const log = logger('webdriverio')
 
+type MobileScrollUntilVisibleOptions = {
+    browser: WebdriverIO.Browser;
+    element: WebdriverIO.Element;
+    maxScrolls: number;
+    scrollDirection: MobileScrollDirection;
+    scrollableElement: WebdriverIO.Element | null;
+};
+
+async function getScrollableElement({
+    browser,
+    options
+}: {
+    browser: WebdriverIO.Browser,
+    options?: MobileScrollIntoViewOptions
+    }): Promise<WebdriverIO.Element | null> {
+    if (options?.scrollableElement) {
+        return options?.scrollableElement
+    }
+    const defaultAndroidSelector = '//android.widget.ScrollView'
+    const defaultIosSelector = '-ios predicate string:type == "XCUIElementTypeApplication"'
+    const selector = browser.isIOS
+        ? // For iOS, we need to find the application element, if we can't find it, we should throw an error
+        defaultIosSelector
+        : // There is always a scrollview for Android or, if this fails we should throw an error
+        defaultAndroidSelector
+    // Not sure why we need to do this, but it seems to be necessary
+    const scrollableElements = (await browser.$$(
+        selector
+    )) as unknown as WebdriverIO.Element[]
+
+    if (scrollableElements.length > 0) {
+        return scrollableElements[0]
+    }
+
+    throw new Error(
+        `Default scrollable element '${browser.isIOS ? defaultIosSelector : defaultAndroidSelector}' not found.`
```

Comment:
What should the user do in this case? Is this a common scenario? If so, we should add a recommendation on what a user should try next.

Reply:
Good suggestion, I'll update the error